airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -104,7 +104,9 @@ def get_iam_token(self, conn: Connection) -> tuple[str, str, int]:
         port = conn.port or 5439
         # Pull the custer-identifier from the beginning of the Redshift URL
         # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
-        cluster_identifier = conn.extra_dejson.get("cluster_identifier", conn.host.split(".")[0])
+        cluster_identifier = conn.extra_dejson.get("cluster_identifier", None)
+        if conn.host:
+            cluster_identifier = conn.extra_dejson.get("cluster_identifier", conn.host.split(".")[0])
```

Comment:
```suggestion
        cluster_identifier = conn.extra_dejson.get("cluster_identifier")
        if not cluster_identifier and conn.host:
            cluster_identifier = conn.host.split(".")[0]
```
Maybe this?

Reply:
[Done](https://github.com/apache/airflow/pull/31567/commits/3bae7f3dc6f5cf179adfa41b2826f232763add05)